src/db_impl.cc

```diff
@@ -632,19 +641,18 @@ void TitanDBImpl::OnFlushCompleted(const FlushJobInfo& flush_job_info) {
 
 void TitanDBImpl::OnCompactionCompleted(
     const CompactionJobInfo& compaction_job_info) {
+  if (!compaction_job_info.status.ok()) {
+    // TODO: Clean up blob file generated by the failed compaction.
+    return;
+  }
   std::map<uint64_t, int64_t> blob_files_size;
   std::set<uint64_t> outputs;
   std::set<uint64_t> inputs;
-  auto calc_bfs = [&compaction_job_info, &blob_files_size, &outputs, &inputs](
-                      const std::vector<std::string>& files, int coefficient,
+  auto calc_bfs = [&](const std::vector<std::string>& files, int coefficient,
                       bool output) {
     for (const auto& file : files) {
       auto tp_iter = compaction_job_info.table_properties.find(file);
       if (tp_iter == compaction_job_info.table_properties.end()) {
-        if (output) {
```

Comment:
why remove it?

Reply:
Seems to me it if fine if compaction output file don't have table property. But let me add a warn level log here.